### PR TITLE
docs(client): build API reports from rollups

### PR DESCRIPTION
**Pull request by jason-ha**
in all packages with legacy exports

Split api-extractor report generation to `build:api-report` task (when it is not done as part of `build:docs`).
+ add `api-extractor-model.esm.json` as common config for all `build:docs` that are just concerned with *.api.json generation.
+ rename shared report configs to say `api-extractor-report` to indicate that is what they cover.

Changes for each impacted package:
1. new `api-extractor/api-extractor.current.json` using `public.d.ts` entrypoint for `*.(public|beta).api.md` generation (there are no `@beta` APIs currently)
2. `api-extractor.json` just used for doc model `*.api.json`generation
3. `api-extractor/api-extractor.legacy.json` edited for common `api-extractor-report.esm.legacy.json` rename
4. package.json - script split up

\* client-utils is special per browser/node entrypoints

Fixes build defect where packages with legacy exports would double generate `*.api.json` doc model and `tsdoc-metadata.json` from two different commands. (Generation from the legacy report step was extraneous and luckily always the same for packages where that happened.)

`*.(public|beta).api.md` changes show API removal of types that are not real exports, but collateral from api-extactor previously keying off of internal entrypoint (index.d.ts).

### CODE REVIEW

File: azure/packages/azure-service-utils/api-report/azure-service-utils.beta.api.md [hunk 1]
```diff
@@ -6,6 +6,4 @@
 
 export { IUser }
 
-export { ScopeType }
```
Comment (Josmithr) [inline azure/packages/azure-service-utils/api-report/azure-service-utils.beta.api.md:9]: Where did this change come from?

File: azure/packages/azure-service-utils/package.json [hunk 2]
```diff
@@ -51,11 +51,12 @@
...
 		"build": "fluid-build . --task build",
+		"build:api-reports": "concurrently \"npm:build:api-reports:*\"",
+		"build:api-reports:current": "api-extractor run --local --config api-extractor/api-extractor.current.json",
+		"build:api-reports:legacy": "api-extractor run --local --config api-extractor/api-extractor.legacy.json",
 		"build:commonjs": "fluid-build . --task commonjs",
 		"build:compile": "fluid-build . --task compile",
-		"build:docs": "concurrently \"npm:build:docs:*\"",
-		"build:docs:current": "api-extractor run --local",
-		"build:docs:legacy": "api-extractor run --local --config api-extractor/api-extractor.legacy.json",
+		"build:docs": "api-extractor run --local",
```
Comment (Josmithr) [inline azure/packages/azure-service-utils/package.json:59]: Not for this PR, but I would like to rename this script. I think the name is misleading. This is generating a docs model, not actually building any documentation. Ideally we could reclaim this for things like `build:readme` and other actual documentation generation.

**Review by Josmithr — APPROVED** [review]
I really like the new naming scheme. Much easier to reason about.

File: azure/packages/azure-service-utils/api-report/azure-service-utils.beta.api.md (hunk 1 above)
Comment (jason-ha) [reply azure/packages/azure-service-utils/api-report/azure-service-utils.beta.api.md:9]: It is called out - the last line of the description. ;)

File: azure/packages/azure-service-utils/package.json (hunk 2 above)
Comment (jason-ha) [reply azure/packages/azure-service-utils/package.json:59]: 👍 The `api-extractor:*` script entries also need to be renamed.

Comment (Josmithr) [reply azure/packages/azure-service-utils/package.json:59]: ...I swear that wasn't there when I read it..... I am suspicious......
image

**Pull request merged**